odoo/addons/base/tests/test_ir_attachment.py

```diff
@@ -75,53 +75,58 @@ def test_06_linked_record_permission(self):
         env = self.env(user=self.env.ref('base.user_demo'))
         Attachment = env['ir.attachment']
 
-        a1 = Attachment.create({'name': 'a1'})
-        log = env['res.users.log'].create({})
-        with self.assertRaises(AccessError):
-            log.write({})
-        vals = {'name': 'attach', 'res_id': a1.id, 'res_model': 'ir.attachment'}
-        a2 = Attachment.create(vals)
+        # create a record with an attachment
+        record1 = Attachment.create({'name': 'record1'})
+        vals = {'name': 'attach', 'res_id': record1.id, 'res_model': record1._name}
+        a1 = Attachment.create(vals)
 
-        # remove access to linked record a1
+        # remove all access rights to record1
         rule = self.env['ir.rule'].create({
-            'name': 'test_rule', 'domain_force': "[('id', '!=', %s)]" % a1.id,
-            'model_id': self.env.ref('base.model_ir_attachment').id,
+            'name': 'test_rule', 'domain_force': "[('id', '!=', %s)]" % record1.id,
+            'model_id': self.env['ir.model']._get_id(record1._name),
```

Comment:
The goal was not to rewrite the entire test in 13, it was to add checks for previously untested behaviours.